crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -127,6 +134,76 @@ impl Fulfillment {
         };
         Some(eth::TokenAmount(amount))
     }
+
+    /// Returns the surplus denominated in the surplus token.
+    ///
+    /// The surplus token is the buy token for a sell order and sell token for a
+    /// buy order.
+    pub fn surplus(
+        &self,
+        sell_amount: eth::U256,
+        buy_amount: eth::U256,
+        prices: ClearingPrices,
+    ) -> Result<eth::U256, Error> {
```

Comment:
The signature looks a bit odd now. Since `Trade` should already know its own limit price and executed amount we should only have to pass in a clearing price to get a surplus.
This is because we want to compute surplus of a trade compared to some price other than the limit price as well for the different fee mechanisms.
Maybe this would be slightly nicer with 2 functions like `Trade::surplus_over_reference_price()` which takes the reference `buy_price` and `sell_price` (currently `sell_amount` and `buy_amount` respectively) and `Trade::surplus_over_limit_price()` which simply calls the other function with the orders limit price.

Reply:
I renamed the function to `surplus_over_reference_price`.